### DISCUSSION TITLE
Container Image Registry Settings - credentials

### DISCUSSION THREAD

**Discussion by cyrus-mc** [bottlerocket-os/bottlerocket#2365] [General]
I am trying to set container registry credentials via `userdata`. The docs say do you the `apiclient` but since I need the credentials in place for the bootstrap container I can't set it once the system is up and running.

My userdata contains:

```
[settings.container-registry.credentials]
[
  {
    "registry": "ghcr.io",
    "auth": "xxxxxxx"
  }
]
```

Where xxx is the base64 encoded username:github_token.

The system doesn't boot so I am assuming it is hanging on trying to pull the bootstrap container. Given the box doesn't boot I am somewhat stuck at trying to troubleshoot. Any assistance would be appreciated.

**Comment by cyrus-mc** [comment]
So I tested a few more things. I took away the bootstrap container (so no need to pull) and the image still failed to boot successfully. I then removed the `settings.container-registry.credentials` block completely and it booted fine. So it appears my format above for specifying the creds is incorrect.

**Reply by cyrus-mc** [reply]
Ok, think I figured it out. Just needed to read a little closer and look to the example for mirrors above.

```
[[settings.container-registry.credentials]]
"registry" = "ghcr.io"
"auth" = "xxxx"
```

Is the format. And then just repeat this block for additional registries.

**Reply by mchaker** [reply]
@cyrus-mc , I'm glad you found a solution.
